exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -84,7 +83,6 @@ public static PrometheusHttpServerBuilder builder() {
               .port(port)
               .executorService(executor)
               .registry(prometheusRegistry)
-              .defaultHandler(new MetricsHandler(prometheusRegistry))
```

Comment:
Let's add a configuration option for this to PrometheusHttpServerBuilder:
```
public PrometheuesHttpServerBuilder setDefaultHandler(com.sun.net.httpserver.HttpHandler defaultHandler)
```
This will allow users to depend on this behavior to restore the current behavior (although it requires programmatic configuration) while aligning with standard prometheus client library behavior.

Reply:
I pushed a commit to add this but it broke the animalsniffer check because `com.sun.net.httpserver.HttpHandler` isn't available in the android API. Prometheus exporter doesn't need to support android environments, and doesn't work on them currently despite animalsniffer passing (animalsniffer apparently can't detect usage of unsupported APIs in dependencies). 

I opened #6478 to separately address this. Can merge that, then merge `main` into this branch and merge this PR.